From ce220825972667a82e5655410de971bf5ae79937 Mon Sep 17 00:00:00 2001
From: Vasyl Yushchyshen <vasyl.yushchyshen@ti.com>
Date: Wed, 18 Feb 2015 09:04:23 -0800
Subject: [PATCH 093/111] omap: platform: ion: Change OMAP ION board data
 structure
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

According to changes in the ION core, omap4_ion_heaps[] is placed outside
the omap4_ion_data structure. The pointer used instead.
The same change in OMAP5 code.

Change-Id: I070ebccbb75a506196f584277d437097f61f3873
Signed-off-by: Vasyl Yushchyshen <vasyl.yushchyshen@globallogic.com>
Signed-off-by: D. Andrei Măceș <dmaces@nd.edu>
---
 arch/arm/mach-omap2/omap4_ion.c | 54 +++++++++++++++++++++--------------------
 1 file changed, 28 insertions(+), 26 deletions(-)

diff --git a/arch/arm/mach-omap2/omap4_ion.c b/arch/arm/mach-omap2/omap4_ion.c
index 9ae6d74..a3f09d86 100644
--- a/arch/arm/mach-omap2/omap4_ion.c
+++ b/arch/arm/mach-omap2/omap4_ion.c
@@ -62,34 +62,36 @@ static size_t omap4_ion_rpmsg_cma_pages_count;
 static struct page* omap4_ion_rpmsg_cma_pages;
 #endif
 
+static struct ion_platform_heap omap4_ion_heaps[] = {
+	{
+		.type = ION_HEAP_TYPE_CARVEOUT,
+		.id = OMAP_ION_HEAP_SECURE_INPUT,
+		.name = "secure_input",
+	},
+	{	.type = OMAP_ION_HEAP_TYPE_TILER,
+		.id = OMAP_ION_HEAP_TILER,
+		.name = "tiler",
+	},
+	{
+		.type = OMAP_ION_HEAP_TYPE_TILER,
+		.id = OMAP_ION_HEAP_NONSECURE_TILER,
+		.name = "nonsecure_tiler",
+	},
+	{
+		.type = ION_HEAP_TYPE_SYSTEM,
+		.id = OMAP_ION_HEAP_SYSTEM,
+		.name = "system",
+	},
+	{
+		.type = OMAP_ION_HEAP_TYPE_TILER_RESERVATION,
+		.id = OMAP_ION_HEAP_TILER_RESERVATION,
+		.name = "tiler_reservation",
+	},
+};
+
 static struct ion_platform_data omap4_ion_data = {
 	.nr = 5,
-	.heaps = {
-		{
-			.type = ION_HEAP_TYPE_CARVEOUT,
-			.id = OMAP_ION_HEAP_SECURE_INPUT,
-			.name = "secure_input",
-		},
-		{	.type = OMAP_ION_HEAP_TYPE_TILER,
-			.id = OMAP_ION_HEAP_TILER,
-			.name = "tiler",
-		},
-		{
-			.type = OMAP_ION_HEAP_TYPE_TILER,
-			.id = OMAP_ION_HEAP_NONSECURE_TILER,
-			.name = "nonsecure_tiler",
-		},
-		{
-			.type = ION_HEAP_TYPE_SYSTEM,
-			.id = OMAP_ION_HEAP_SYSTEM,
-			.name = "system",
-		},
-		{
-			.type = OMAP_ION_HEAP_TYPE_TILER_RESERVATION,
-			.id = OMAP_ION_HEAP_TILER_RESERVATION,
-			.name = "tiler_reservation",
-		},
-	},
+	.heaps = omap4_ion_heaps,
 };
 
 static struct omap_ion_platform_data omap4_ion_pdata = {
-- 
2.7.4

